db/blob/blob_source.cc

```diff
@@ -152,6 +153,142 @@ Status BlobSource::GetBlob(const ReadOptions& read_options,
   return s;
 }
 
+void BlobSource::MultiGetBlob(
+    const ReadOptions& read_options,
+    const autovector<std::reference_wrapper<const Slice>>& user_keys,
+    uint64_t file_number, uint64_t file_size,
+    const autovector<uint64_t>& offsets,
+    const autovector<uint64_t>& value_sizes, autovector<Status*>& statuses,
+    autovector<PinnableSlice*>& blobs, uint64_t* bytes_read) {
+  size_t num_blobs = user_keys.size();
+  assert(num_blobs > 0);
+  assert(num_blobs <= MultiGetContext::MAX_BATCH_SIZE);
+  assert(num_blobs == offsets.size());
+  assert(num_blobs == value_sizes.size());
+  assert(num_blobs == statuses.size());
+  assert(num_blobs == blobs.size());
+
+#ifndef NDEBUG
+  for (size_t i = 0; i < offsets.size() - 1; ++i) {
+    assert(offsets[i] <= offsets[i + 1]);
+  }
+#endif  // !NDEBUG
+
+  using Mask = uint64_t;
+  Mask cache_hit_mask_ = 0;
+
+  Status s;
+  uint64_t total_bytes = 0;
+  const OffsetableCacheKey base_cache_key(db_id_, db_session_id_, file_number,
+                                          file_size);
+
+  if (blob_cache_) {
+    size_t cached_blob_count = 0;
+    for (size_t i = 0; i < num_blobs; ++i) {
+      CachableEntry<std::string> blob_entry;
+      const CacheKey cache_key = base_cache_key.WithOffset(offsets[i]);
+      const Slice key = cache_key.AsSlice();
+
+      s = GetBlobFromCache(key, &blob_entry);
+      if (!s.ok()) {
```

Comment:
`GetBlobFromCache` not returning OK is not a failure, right? Seems to me it returns `NotFound` when the blob is not in the cache, in which case we would want to proceed to read it from the file.

Reply:
Yeah. But I only update its status here.  What's the reason we use bitmask.

Reply:
I still feel it would be safer if we didn't update the status to `NotFound` though, like you do when retrieving a single blob: https://github.com/facebook/rocksdb/blob/main/db/blob/blob_source.cc#L97-L108